progetto/ServerProject/src/server/Server.java

```diff
@@ -36,28 +38,47 @@
 import java.security.InvalidAlgorithmParameterException;
 import java.security.InvalidKeyException;
 import java.security.NoSuchAlgorithmException;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Properties;
 import java.util.concurrent.Callable;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentSkipListMap;
 
 public class Server implements ServerInterface,Callable<Integer> {
-    //STRUTTURE DATI
+
+    /* topic and message management fields */
+    //todo se qualcuno trova un nome migliore cambitelo quello che ci ho messo fa schifo
+    private ConcurrentSkipListMap<String,ConcurrentLinkedQueue<Integer>> topicClientList;                 // topic -> lista idAccount    -   PUNTI 1 e 2
+    //invece di rompere sempre le balle all'account manager dandogli id dello stub si potrebbe salvare direttamente lo stub
```

Comment:
secondo me è una ridondanza che è meglio non dovere gestire perchè
1) quando avviene un aggiornamento dello stub su un account bisogna gestire  la consistenza dei dati tra le due strutture dati(questa lista e l'account monitor)
2)mi sembra un overkill delle performance
3)l'account monitor ha un reader writer lock quindi come dicevo continua a sembrarmi un approccio troppo incentrato sulle performance quando in realtà non ce n è bisogno . Anche se il messaggio arrivasse un secondo dopo(improbabile) non è la fine del mondo